### ISSUE TITLE
Update to libssh2 1.9.0

### ISSUE THREAD

**Issue by Jan-E** [winlibs/libssh2#3]
@weltling @cmb69 
I had some trouble updating to libssh2 1.9.0. The changes in the project files are not obvious. You'll have to add three new 'existing items': blf.h, bcrypt_pbkdf.c and blowfish.c. Like this:

![libssh2](https://user-images.githubusercontent.com/1757825/60383608-63ee7800-9a73-11e9-9c64-95efca6175c6.png)

Alternative route: create new project files by converting the libssh2.dsw and libssh2.dsp that are in the release tarballs of libssh2 1.9.0

**Comment by weltling** [comment]
Thanks for the heads up and the tips, @Jan-E. We'll see to upgrade it for the next or the one after it. If you have a PR with the fix, also for the lower VS versions, please file it.

Thanks.

**Comment by Jan-E** [comment]
I do not have a PR, because I ended up converting the dsw/dsp. When I saw that this solved my initial compile errors, I investigated a little further and found out that blf.h had to be added as well. I tested this on the VS16 project file (with my own deps-path in stead of ..\deps).

**Comment by Jan-E** [comment]
I do not have a PR, because I ended up converting the dsw/dsp. When I saw that this solved my initial compile errors, I investigated a little further and found out that blf.h had to be added as well. I tested this on the VS16 project file (with my own deps-path in stead of ..\deps).

**Comment by Jan-E** [comment]
See https://libssh2.org/mail/libssh2-devel-archive-2019-06/0006.shtml and followup messages.

**Comment by weltling** [comment]
Thanks for more info. So it seems these files are not just missing in the solution, they're missing in the tarball. Lets see, probably some other libssh2 release will happen.

Thanks.

**Comment by Jan-E** [comment]
No, for Cmake some files were missing. For existing project files everything is there.

**Comment by cmb69** [comment]
Thanks, @Jan-E. I can confirm that we basically need the following patch:
````.patch
 win32/libssh2_dll.vs16.vcxproj | 3 +++
 win32/libssh2_lib.vs16.vcxproj | 3 +++
 2 files changed, 6 insertions(+)

diff --git a/win32/libssh2_dll.vs16.vcxproj b/win32/libssh2_dll.vs16.vcxproj
index 4325204..fd4356d 100644
--- a/win32/libssh2_dll.vs16.vcxproj
+++ b/win32/libssh2_dll.vs16.vcxproj
@@ -275,6 +275,8 @@
   </ItemDefinitionGroup>
   <ItemGroup>
     <ClCompile Include="..\src\agent.c" />
+    <ClCompile Include="..\src\bcrypt_pbkdf.c" />
+    <ClCompile Include="..\src\blowfish.c" />
     <ClCompile Include="..\src\channel.c" />
     <ClCompile Include="..\src\comp.c" />
     <ClCompile Include="..\src\crypt.c" />
@@ -300,6 +302,7 @@
   </ItemGroup>
   <ItemGroup>
     <ClInclude Include="..\include\libssh2.h" />
+    <ClInclude Include="..\src\blf.h" />
     <ClInclude Include="libssh2_config.h" />
     <ClInclude Include="..\include\libssh2_priv.h" />
     <ClInclude Include="..\include\libssh2_sftp.h" />
diff --git a/win32/libssh2_lib.vs16.vcxproj b/win32/libssh2_lib.vs16.vcxproj
index c789dfd..41c6382 100644
--- a/win32/libssh2_lib.vs16.vcxproj
+++ b/win32/libssh2_lib.vs16.vcxproj
@@ -199,6 +199,8 @@
   </ItemDefinitionGroup>
   <ItemGroup>
     <ClCompile Include="..\src\agent.c" />
+    <ClCompile Include="..\src\bcrypt_pbkdf.c" />
+    <ClCompile Include="..\src\blowfish.c" />
     <ClCompile Include="..\src\channel.c" />
     <ClCompile Include="..\src\comp.c" />
     <ClCompile Include="..\src\crypt.c" />
@@ -224,6 +226,7 @@
   </ItemGroup>
   <ItemGroup>
     <ClInclude Include="..\include\libssh2.h" />
+    <ClInclude Include="..\src\blf.h" />
     <ClInclude Include="libssh2_config.h" />
     <ClInclude Include="..\include\libssh2_priv.h" />
     <ClInclude Include="..\include\libssh2_sftp.h" />
````

**Comment by Jan-E** [comment]
OK. I am closing this issue and leave it up to you to update in due time.

**Comment by Jan-E** [comment]
In the mean time LibSSH2 1.10.0 has been released, with some fixes and support for OpenSSL 3.0 as well. Is not it time to update?

**Comment by cmb69** [comment]
Yeah, I've noticed that, but I didn't see any security fixes, so the update should only be for PHP-8.1 for now. I'll see to it.
